server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java

```diff
@@ -0,0 +1,309 @@
+                try {
+                    Mappings mappings = virtualTableStore.getMappings(recipient.getLocalPart(), recipient.getDomain());
+
+                    if (mappings != null) {
+                        List<MailAddress> newMailAddresses = handleMappings(mappings, mail.getSender(), recipient, mail.getMessage(), mailetContext, domainList);
+                        return new MappingData(Optional.of(newMailAddresses), Optional.<List<MailAddress>>absent());
+                    }
+                } catch (ErrorMappingException e) {
+                    mailetContext.log(LogLevel.INFO, "Error while process mail.", e);
+                    return new MappingData(Optional.<List<MailAddress>>absent(), Optional.<List<MailAddress>>of(originMailAddresses));
+                } catch (RecipientRewriteTableException e) {
+                    mailetContext.log(LogLevel.INFO, "Error while process mail.", e);
+                    return new MappingData(Optional.<List<MailAddress>>absent(), Optional.<List<MailAddress>>of(originMailAddresses));
+                } catch (MessagingException e) {
+                    mailetContext.log(LogLevel.INFO, "Error while process mail.", e);
+                    return new MappingData(Optional.<List<MailAddress>>absent(), Optional.<List<MailAddress>>of(originMailAddresses));
+                }
+                return new MappingData(Optional.<List<MailAddress>>of(originMailAddresses), Optional.<List<MailAddress>>absent());
+            }
+        };
+    }
+
+    private Function<Mapping, Mapping> appendDefaultDomain(final String defaultDomain) {
+        return new Function<Mapping, Mapping>() {
+            @Override
+            public Mapping apply(Mapping address) {
+                return address.appendDomain(defaultDomain);
+            }
+        };
+    }
+
+    private Predicate<MailAddress> isLocalServer(final MailetContext context) {
+        return new Predicate<MailAddress>() {
+            @Override
+            public boolean apply(MailAddress mailAddress) {
+                return context.isLocalServer(mailAddress.getDomain());
+            }
+        };
+    }
+
+    private Predicate<MailAddress> isNotLocalServer(final MailetContext context) {
+        return new Predicate<MailAddress>() {
+            @Override
+            public boolean apply(MailAddress mailAddress) {
+                return !context.isLocalServer(mailAddress.getDomain());
+            }
+        };
+    }
+
+    private void forwardToRemoteAddress(MailAddress sender, MailAddress recipient, MimeMessage message, ImmutableList<MailAddress> mailAddresses, final MailetContext context) throws MessagingException {
+        ImmutableList<MailAddress> remoteAddress = FluentIterable.from(mailAddresses)
+            .filter(isNotLocalServer(context))
+            .toList();
+
+        if (!remoteAddress.isEmpty()) {
+            StringBuilder logBuffer = new StringBuilder(128);
+            try {
```

Comment:
proposal to make the code smoothier, and improve logging : 

```
try {
    context.sendMail(sender, remoteAddress, message);
    context.log(LogLevel.INFO, "Mail for " + recipient + " forwarded to " + remoteAddress);
} catch (MessagingException ex) {
    context.log(LogLevel.WARN, "Error forwarding mail to " + remoteAddress);
}
```

6 lines instead of 12.